website/blog/2021-09-05-2.4.0.md

```diff
@@ -0,0 +1,248 @@
+}
+
+// Prettier 2.3
+SyntaxError: Declaration expected. (5:9)
+  3 |
+  4 |   // This is a static block:
+> 5 |   static {
+    |         ^
+  6 |     if (someCondition()) {
+  7 |       Foo.count++;
+  8 |     }
+
+// Prettier 2.4
+class Foo {
+  static count = 0;
+
+  // This is a static block:
+  static {
+    if (someCondition()) {
+      Foo.count++;
+    }
+  }
+}
+
+```
+
+### HTML
+
+#### Replace `jsxBracketSameLine` option with `bracketSameLine` option ([#11006](https://github.com/prettier/prettier/pull/11006) by [@kurtztech](https://github.com/kurtztech))
+
+Deprecate the `jsxBracketSameLine` option in favour of new `bracketSameLine` option that will work for HTML, Angular, Vue, and JSX.
+
+<!-- prettier-ignore -->
+```html
+<!-- Input -->
+<div id="foo-bar-baz"
+  class="bar-foo-baz"
+  title="a sample title"
+  data-foo="bar"
+  data-bar="baz">lorem ipsum dolor sit amet</div>
+
+<!-- Prettier 2.3 -->
+<div
+  id="foo-bar-baz"
+  class="bar-foo-baz"
+  title="a sample title"
+  data-foo="bar"
+  data-bar="baz"
+>
+  lorem ipsum dolor sit amet
+</div>
+
+<!-- Prettier 2.4 -->
+<!-- Options: `{bracketSameLine: true}` -->
+<div
+  id="foo-bar-baz"
+  class="bar-foo-baz"
+  title="a sample title"
+  data-foo="bar"
+  data-bar="baz">
+  lorem ipsum dolor sit amet
+</div>
+```
+
+## Other Changes
+
+### JavaScript
+
+#### Support parenthesized tagged template literals for styled components ([#11246](https://github.com/prettier/prettier/pull/11246) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```js
+// Input
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position:   fixed;
+  color: red;
+  `;
+
+// Prettier 2.4
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position:   fixed;
+  color: red;
+  `;
+
+// Prettier 2.3
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position: fixed;
+    color: red;
+  `;
+
+```
+
+#### Count correctly regex literals in method chain arguments ([#11299](https://github.com/prettier/prettier/pull/11299) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```js
+// Input
+foo1(/𠮟𠮟𠮟/).foo2(bar).foo3(baz);
+
+foo1(/叱叱叱/).foo2(bar).foo3(baz);
+
+// Prettier 2.3
+foo1(/𠮟𠮟𠮟/)
+  .foo2(bar)
+  .foo3(baz);
+
+foo1(/叱叱叱/).foo2(bar).foo3(baz);
+
+// Prettier 2.4
+foo1(/𠮟𠮟𠮟/)
+  .foo2(bar)
+  .foo3(baz);
+
+foo1(/叱叱叱/)
+  .foo2(bar)
+  .foo3(baz);
+
+```
+
+#### Support hack-style pipeline proposal ([#11335](https://github.com/prettier/prettier/pull/11335) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Support new [hack-style pipeline proposal](https://github.com/js-choi/proposal-hack-pipes). We chose `%` as the topic token, following the official explainer.
```

Comment:
```suggestion
The proposed [hack-style pipeline syntax](https://github.com/js-choi/proposal-hack-pipes) is now supported by Prettier. We are using `%` as the topic token, following the official explainer — although this may change in future releases as the proposal progresses.
```